plaidml2/bridge/keras/backend_test.py

```diff
@@ -1599,27 +1657,26 @@ def testConvParameterRankExceptions(self):
     @unittest.skipIf(
         os.environ.get("USE_STRIPE", "0") == "1",
         "Stripe does not correctly validate assignment ops")
-    @unittest.skip('TODO: convert to EDSL')
     def testAssignmentExceptions(self):
-        A = pkb.variable(m(5, 1))
-        B = pkb.variable(m(1, 5))
-        f = """function (A[L, M], B[M, N]) -> (O) {
-                   O[i, k: L, N] = =(A[i, j] * B[j, k]);
-               }"""
-        # A * B has each entry a "sum" of exactly one product, and so assignment
-        # is valid and should be the same as + aggregation.
-        O = tile.Operation(f, [('A', A), ('B', B)],
-                           [('O', tile.Shape(A.shape.dtype,
-                                             (A.shape.dims[0], B.shape.dims[1])))]) \
-                .sole_output().eval()
-        npt.assert_allclose(O, np.dot(m(5, 1), m(1, 5)))
-        # B * A sums multiple products into one output entry, and so assignment
-        # is not valid and should raise a multiple assignment error.
-        with self.assertRaises(plaidml.exceptions.Unknown) as cm:
-            tile.Operation(f, [('A', B), ('B', A)],
-                           [('O', tile.Shape(A.shape.dtype,
-                                             (A.shape.dims[0], B.shape.dims[1])))]) \
-                .sole_output().eval()
+        A = pkb.variable(m(5, 1)).tensor
+        B = pkb.variable(m(1, 5)).tensor
+        L = edsl.TensorDim()
+        M = edsl.TensorDim()
+        N = edsl.TensorDim()
+        i = edsl.TensorIndex()
+        j = edsl.TensorIndex()
+        k = edsl.TensorIndex()
+        A.bind_dims(L, M)
+        B.bind_dims(M, N)
+        O = edsl.TensorOutput(L, N)
+        O[i, j] = A[i, k] * B[k, j]
+        OutputAB = pkb._KerasNode('assignment_non_exception', name='opAB', tensor=O)
+        npt.assert_allclose(pkb.eval(OutputAB), np.dot(m(5, 1), m(1, 5)))
+        with self.assertRaises(RuntimeError) as cm:
+            O = edsl.TensorOutput(L, N)
+            O[i, j] = B[i, j] * A[j, k]
+            OutputBA = pkb._KerasNode('assignment_exception', name='opBA', tensor=O)
+            OutputBA.eval()
         self.assertTrue("Multiple assignment" in str(cm.exception))
```

Comment:
This "Multiple assignment" line probably needs to be indented, I think `cm` will be out of scope otherwise